x/ibc/testing/chain.go

```diff
@@ -0,0 +1,373 @@
+package testing
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	abci "github.com/tendermint/tendermint/abci/types"
+	"github.com/tendermint/tendermint/crypto"
+	"github.com/tendermint/tendermint/crypto/secp256k1"
+	lite "github.com/tendermint/tendermint/lite2"
+	tmtypes "github.com/tendermint/tendermint/types"
+
+	"github.com/cosmos/cosmos-sdk/simapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	connectiontypes "github.com/cosmos/cosmos-sdk/x/ibc/03-connection/types"
+	channeltypes "github.com/cosmos/cosmos-sdk/x/ibc/04-channel/types"
+	ibctmtypes "github.com/cosmos/cosmos-sdk/x/ibc/07-tendermint/types"
+	commitmenttypes "github.com/cosmos/cosmos-sdk/x/ibc/23-commitment/types"
+	host "github.com/cosmos/cosmos-sdk/x/ibc/24-host"
+	"github.com/cosmos/cosmos-sdk/x/ibc/keeper"
+)
+
+const (
+	// Default params used to create a TM client
+	TrustingPeriod  time.Duration = time.Hour * 24 * 7 * 2
+	UnbondingPeriod time.Duration = time.Hour * 24 * 7 * 3
+	MaxClockDrift   time.Duration = time.Second * 10
+
+	ConnectionVersion = "1.0"
+	ChannelVersion    = "1.0"
+
+	ConnectionIDPrefix = "connectionID"
+	ChannelIDPrefix    = "channelID"
+	PortIDPrefix       = "portID"
```

Comment:
what are these for?

Reply:
naming prefixes when creating connections and channels. This is just a substitution to `testPort1` etc

Reply:
alternatively these could be randomly generated, but I figured something readable is useful ie `channel/channelID0/port/portID0`
`channel/channelID1/port/portID1`